command/setup_vault.go

```diff
@@ -0,0 +1,675 @@
+	return nil
+}
+
+// askQuestion asks question to user until they provide a valid response.
+func (s *SetupVaultCommand) askQuestion(question string) bool {
+	for {
+		answer, err := s.Ui.Ask(s.Colorize().Color(fmt.Sprintf("[?] %s", question)))
+		if err != nil {
+			if err.Error() != "interrupted" {
+				s.Ui.Output(err.Error())
+				os.Exit(1)
+			}
+			os.Exit(0)
+		}
+
+		switch strings.TrimSpace(strings.ToLower(answer)) {
+		case "", "y", "yes":
+			return true
+		case "n", "no":
+			return false
+		default:
+			s.Ui.Output(fmt.Sprintf(`%q is not a valid response, please answer "yes" or "no".`, answer))
+			continue
+		}
+	}
+}
+
+func (s *SetupVaultCommand) handleNo() {
+	s.Ui.Warn(`
+By answering "no" to any of these questions, you are risking an incorrect Consul
+cluster configuration. Nomad workloads with Workload Identity will not be able
+to authenticate unless you create missing configuration yourself.
+ `)
+
+	exitCode := 0
+	if s.autoYes || s.askQuestion("Remove everything this command creates? [Y/n]") {
+		exitCode = s.removeConfiguredComponents()
+	}
+
+	s.Ui.Output(s.Colorize().Color(`
+Consul cluster has [bold][underline]not[reset] been configured for authenticating Nomad tasks and
+services using workload identitiies.
+
+Run the command again to finish the configuration process.`))
+	os.Exit(exitCode)
+}
+
+func (s *SetupVaultCommand) removeConfiguredComponents() int {
+	s.vClient.SetNamespace(s.ns)
+	exitCode := 0
+	componentsToRemove := map[string]string{}
+
+	if s.policyExists() {
+		componentsToRemove["Policy"] = consulPolicyName
+	}
+	if s.roleExists() {
+		componentsToRemove["Role"] = consulRoleTasks
+	}
+	if s.jwtEnabled() {
+		componentsToRemove["JWT Credential backend and its configuration"] = ""
+	}
+	if s.namespaceExists(s.ns) {
+		componentsToRemove["Namespace"] = s.ns
+	}
+	if exitCode != 0 {
+		return exitCode
+	}
+
+	q := `The following items will be deleted:
+%s`
+	if len(componentsToRemove) == 0 {
+		s.Ui.Output("Nothing to delete.")
+		return 0
+	}
+
+	if !s.autoYes {
+		s.Ui.Warn(fmt.Sprintf(q, printMapOfStrings(componentsToRemove)))
+	}
+
+	if s.autoYes || s.askQuestion("Remove all the items listed above? [Y/n]") {
+
+		if policy, ok := componentsToRemove["Policy"]; ok {
+			err := s.vClient.Sys().DeletePolicy(policy)
+			if err != nil {
+				s.Ui.Error(fmt.Sprintf("[✘] Failed to delete policy %q: %v", policy, err.Error()))
+				exitCode = 1
+			} else {
+				s.Ui.Info(fmt.Sprintf("[✔] Deleted policy %q.", policy))
+			}
+		}
+
+		if role, ok := componentsToRemove["Role"]; ok {
+			_, err := s.vLogical.Delete(fmt.Sprintf("/auth/jwt/role/%s", role))
+			if err != nil {
+				s.Ui.Error(fmt.Sprintf("[✘] Failed to delete role %q: %v", role, err.Error()))
+				exitCode = 1
+			} else {
+				s.Ui.Info(fmt.Sprintf("[✔] Deleted role %q.", role))
+			}
+		}
+
+		if _, ok := componentsToRemove["JWT Credential backend and its configuration"]; ok {
+			if err := s.vClient.Sys().DisableAuth("jwt"); err != nil {
```

Comment:
Hum...it's probably safe to leave the JWT backend enabled, but we should remove the auth method we create.

Reply:
Huh, I thought maybe if we clean everything then we should clean everything? Credential backends are also namespaced, so for ent this matters I guess?